gym/vector/utils/spaces.py

```diff
@@ -87,3 +89,92 @@ def batch_space_dict(space, n=1):
 
 def batch_space_custom(space, n=1):
     return Tuple(tuple(space for _ in range(n)))
+
+
+@singledispatch
+def iterate(space, items):
+    """Iterate over the elements of a (batched) space.
+
+    Parameters
+    ----------
+    items : samples of `space`
```

Comment:
Nit: those docstrings and the doctest below still have the "items, space" ordering